hathor/event/storage/memory_storage.py

```diff
@@ -12,24 +12,37 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-from typing import Dict, Optional
+from typing import List, Optional
 
 from hathor.event.base_event import BaseEvent
 from hathor.event.storage.event_storage import EventStorage
 
 
 class EventMemoryStorage(EventStorage):
     def __init__(self):
-        self.events: Dict[int, BaseEvent] = {}
+        self._events: List[BaseEvent] = []
+        self._last_event: Optional[BaseEvent] = None
+        self._last_group_id: Optional[int] = None
 
     def save_event(self, event: BaseEvent) -> None:
-        self.events[event.id] = event
+        if event.id != len(self._events):
+            raise ValueError('invalid event.id, ids must be sequential and leave no gaps')
```

Comment:
Should we have a validation like this for the RocksDB storage?
Maybe not based on the number of keys, but on the id of the last event

Reply:
That'd be good.